pyrival/algebra/discrete_log.py

```diff
@@ -1,22 +1,55 @@
+from math import gcd, sqrt
```

Comment:
Sorry, my review process is slow.

I would prefer `x**0.5` in favor of sqrt and writing gcd out explicitly, like [here](https://github.com/cheran-senthil/PyRival/commit/cfc4643742d5c2fe183d095c7fb2d3a18c8f3d60) is better practice long term for copying code in Python 2.

Reply:
From my experience, math.sqrt is both faster and more precise compared to x**0.5. I do however agree with writing out gcd.

Reply:
@cheran-senthil no trouble at all, we have all the time :-)

Don't agree with either of the changes. Not replacing `math.sqrt` with `x**0.5` because of the reason @bjorn-martinsson suggested above and also because it compromises readability. Most importantly, none of these methods are the bottlenecks in this algorithm. No matter how much you optimize `gcd` or `sqrt`, the time saved will only be negligible when compared to the `O(sqrt(m))` dict insertions/lookup that immediately follows. As such I find no good reason to unnecessarily lengthen the implementation.

Note that this was not the case for `extended_gcd` as it was being called `O(sqrt(m))` times. So it made sense to optimize that part.